### ISSUE TITLE
Labels are sometimes cut off in ConfusionMatrix 

### ISSUE THREAD

**Issue by rebeccabilbro** [DistrictDataLabs/yellowbrick#613]
**Describe the bug**
When a dictionary of encoding-to-string mappings is provided to the `ConfusionMatrix` so that the class names will appear as labels, sometimes the class and axis labels get cut off.

![confusion_matrix_iris](https://user-images.githubusercontent.com/8760385/45598773-2cfd1900-b9af-11e8-9d9b-d6e598b24da4.png)

**To Reproduce**

```python
from sklearn.datasets import load_iris
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split as tts

from yellowbrick.classifier import ConfusionMatrix


iris = load_iris()
iris_X = iris.data
iris_y = iris.target
iris_classes = iris.target_names
i_X_train, i_X_test, i_y_train, i_y_test = tts(
    iris_X, iris_y, test_size=0.2
)
model = LogisticRegression()
iris_cm = ConfusionMatrix(
    model, classes=iris_classes,
    label_encoder={0: 'setosa', 1: 'versicolor', 2: 'virginica'}
)
iris_cm.fit(i_X_train, i_y_train)
iris_cm.score(i_X_test, i_y_test)
i = iris_cm.poof()
```

**Desktop (please complete the following information):**
 - OS:  macOS
 - Python Version 3.6
 - Yellowbrick Version 0.8

**Additional context**
Note that the behavior appears to be somewhat different when the code is executed [inside a Jupyter notebook](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/examples/rebeccabilbro/conf_matrix_class_names.ipynb). Discovered in the process of working on #611

**Comment by alexander-ponomaroff** [comment]
Has this issue been fixed? I don't experience any problems with labels when reproducing your result.

**Comment by rebeccabilbro** [comment]
Hi there @alexander-ponomaroff and welcome to Scikit-YB! Yes, as far as I know, this bug is still an issue, though it doesn't manifest in all scenarios -- I was only seeing it outside of Jupyter notebooks, for instance. Would you mind documenting your OS version, Python version, and YB version and also letting us know whether you were using YB inside or outside an ipynb?

**Comment by lwgray** [comment]
I am reopening this issue because during my implementation of plot directive for the ConfusionMatrix Docs #747 I came across it again.  It doesn't appear to be fixed with recent mpl updates.

* OS:  Ubuntu 18.04.1 LTS
* Python Version 3.6.5
* Yellowbrick Version 0.9
* I use Anaconda 

![confusionmatrix-2](https://user-images.githubusercontent.com/1000117/52799546-07506a80-3048-11e9-8f8f-a416321a3aa8.PNG)

**Comment by lwgray** [comment]
I think I might have found a partial solution (also at #768).  I found that adding the keyword `bbox_inches='tight'` to `poof` fixed the cutoff. 

```
iris_cm = ConfusionMatrix(
    model, classes=classes,
    label_encoder={0: 'setosa', 1: 'versicolor', 2: 'virginica'}
)

iris_cm.fit(X_train, y_train)
iris_cm.score(X_test, y_test)

i=iris_cm.poof(bbox_inches='tight')
```

Before `bbox_inches='tight'`
![image](https://user-images.githubusercontent.com/1000117/53255137-05635880-3693-11e9-9766-48fb69c5068e.png)


After `bbox_inches='tight'`
![image](https://user-images.githubusercontent.com/1000117/53255057-d4832380-3692-11e9-99d7-ab22fa573dcf.png)

**Comment by naresh-bachwani** [comment]
I encountered the same problem. The solution by @lwgray  seems to work well on the saved plot but the rendered plot still has label cutoffs. 

**The rendered plot:**
![captured](https://user-images.githubusercontent.com/43993586/59437310-f73a7300-8e0e-11e9-95d9-89fc08d6626b.PNG)

The **saved figure** seems to be fine. 
![saved_figure](https://user-images.githubusercontent.com/43993586/59437340-0a4d4300-8e0f-11e9-895b-2ef2a62a5321.png)
 
**POSSIBLE FIX**
Use `plt.tight_layout` in `finalize`. It also reduces the need to override the **base** `poof` method. My results on adding `plt.tight_layout` in `finalize`. I also removed the overriding `poof` method(from confusion matrix)  for this.
**Rendered plot:**

![captured1](https://user-images.githubusercontent.com/43993586/59438658-8ba5d500-8e11-11e9-9979-8d19cf1d43fc.png)

**Saved image:**

![saved_figure](https://user-images.githubusercontent.com/43993586/59438673-93fe1000-8e11-11e9-80c3-fff5a885f779.png)

The problem with `plt.tight_layout` as specified by @ndanielsen [here](https://github.com/DistrictDataLabs/yellowbrick/pull/640#issuecomment-433766793) is that it reduces the plot size to fit in the ticks and labels. Suppose if the tick labels are increased in length the plot size becomes smaller. An example is given below:

![saved_figure](https://user-images.githubusercontent.com/43993586/59441862-d37b2b00-8e16-11e9-8bc3-47f23bc209df.png).

In the current scenario too, the plot size(in the saved figure) decreases if I increase the tick size.
